### PR TITLE
Remove all instances of the specified route

### PR DESCRIPTION
**Pull request by srvance**
@nathanhammond identified in #36 that there may be a problem with duplicate instances of the same route. The fix is a bit different, so I thought it better as a separate PR than that one because it will have broader impact due to necessary changes in the return type of `hasRoute`.

### CODE REVIEW

**Comment by srvance** [comment]
I changed `hasRoute` to return an array of matching routes. Previously it was only returning the last matching route.

For `add` I chose to replicate the previous behavior of adding the new route under the last matching parent route. I didn't see any point in adding the nested route multiple times. It seemed like making a situation that shouldn't exist worse. I didn't want to add the complexity of choosing the existing duplicate instance that had the most nested routes. That seemed too much complexity for a corner case that shouldn't exist in the first place.

For `remove` I  removed all matching routes. The return condition in the `reduce` is very reliant on order and short-circuit evaluation to be correct. Do you feel this needs commenting?

I'm removing the `WIP` tag, but I would like to wait for #36 to be merged so I can rebase and add some tests in their intersection.

**Comment by srvance** [comment]
@nathanhammond Any feedback on this?

File: lib/helpers/has-route.js [hunk 1]
```diff
@@ -12,7 +12,7 @@ function isTopLevelExpression(path) {
 }
 
 module.exports = function hasRoute(name, routes, identifier) {
```
Comment (nathanhammond) [inline lib/helpers/has-route.js]: My main complaint here is that this helper is named `hasRoute` which implies a boolean return. It never was boolean to begin with, but maybe we can fix that at the same time. Change this to `getRoutes`?

File: lib/ember-router-generator.js [hunk 2]
```diff
@@ -66,11 +64,11 @@ EmberRouterGenerator.prototype.add = function(routeName, options) {
...
   options = options || {};
-  var parent   =  nameParts[0];
-  var name     = parent;
+  var parent = nameParts[0];
+  var name = parent;
   var children = nameParts.slice(1);
-  var path     = hasRoute(parent, routes, options.identifier);
-  var route    = path ? path.node : path;
+  var paths     = hasRoute(parent, routes, options.identifier);
+  var route    = paths.length ? paths[paths.length - 1].node : false;
```
Comment (nathanhammond) [inline lib/ember-router-generator.js]: In this case, even if there are duplicates, we add it to the last one. Does this match the previous behavior of what `hasRoute` returned (first, or last)?

File: lib/ember-router-generator.js [hunk 3]
```diff
@@ -66,11 +64,11 @@ EmberRouterGenerator.prototype.add = function(routeName, options) {
...
 EmberRouterGenerator.prototype._add = function(nameParts, routes, options) {
   options = options || {};
-  var parent   =  nameParts[0];
-  var name     = parent;
+  var parent = nameParts[0];
+  var name = parent;
   var children = nameParts.slice(1);
-  var path     = hasRoute(parent, routes, options.identifier);
-  var route    = path ? path.node : path;
+  var paths     = hasRoute(parent, routes, options.identifier);
```
Comment (nathanhammond) [inline lib/ember-router-generator.js]: Let's finish cleaning up these spaces here like above.

File: lib/ember-router-generator.js (hunk 2 above)
Comment (srvance) [inline lib/ember-router-generator.js]: Yes. Previous behavior was always to work with the last. It was a side effect of the way the AST was being traversed.

File: lib/helpers/has-route.js (hunk 1 above)
Comment (srvance) [inline lib/helpers/has-route.js]: Mind if I use `findRoutes`? I prefer to reserve `get` for accessors.

**Pull request merged**